src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -52,13 +55,14 @@ public function create(string $resourceClass, string $property, array $options =
 
         try {
             [$normalizationGroups, $denormalizationGroups] = $this->getEffectiveSerializerGroups($options, $resourceClass);
-
-            $propertyMetadata = $this->transformReadWrite($propertyMetadata, $resourceClass, $property, $normalizationGroups, $denormalizationGroups);
-            $propertyMetadata = $this->transformLinkStatus($propertyMetadata, $normalizationGroups, $denormalizationGroups);
         } catch (ResourceClassNotFoundException $e) {
-            // No need to check link status if related class is not a resource
+            // hack to allow non-resource classes (actually they should not be supported)
```

Comment:
If this hack is removed, all hell breaks loose, because we're using the `PropertyMetadataFactory` for input/output classes (DTOs). :disappointed:

Reply:
Yes we done it on purpose. It's a good move actually, it allows to retrieve the shape of any PHP structure, regardless if it is mapped with `@ApiProperty` or not.

Reply:
Okay, I'll update the comment. 😎

Reply:
But actually, I can see many problems with using the property metadata system like this. It's simply not designed for non-resources, as you can see here. It'd give the wrong results, which is arguably worse than not using it at all.

Reply:
@dunglas I hope this is okay for now: https://github.com/api-platform/core/compare/075222cd357ad018bc517e954ab46b5b4e9e90b3..731b45f5554d7ec832a33ed59fd1d1fac2230373